### PR TITLE
Code New - Added content for Gitignore

### PR DESCRIPTION
**Pull request by SomeOrdinaryBro**
Added content and created new files under python (topics ill cover next week)

### CODE REVIEW

**Comment by SomeOrdinaryBro** [comment]
Please dont close @mshajid will add more content tonight

if urgent close this PR and drop a txt on Discord. ty

**Comment by mshajid** [comment]
> Please dont close @mshajid will add more content tonight
> 
> if urgent close this PR and drop a txt on Discord. ty

Yes this is noted.

File: Before You Start/4. Git Basics.md [hunk 1]
```diff
@@ -325,8 +327,106 @@ So we brought you up with a (🧔🏽) _**Bob &**_ (👨🏽‍🦲) _**John "GI
 | Project Progress           | 🧔🏽**Bob** and 👨🏽‍🦲**John** continue this collaborative process, creating new branches for other features or bug fixes as needed. Git keeps a clear history of who did what and when. |
 | Backup and Recovery        | Git serves as a backup for their project. If anything goes wrong, they can always revert to a previous commit to recover their work. |
 
-<!-- I think this is enough Git as beginners will not be forking and creating PR's, So going to start with "Setting up git with github" below -->
----
+# Understanding .gitignore for Beginners
+
+A `.gitignore` file is like a list of instructions you give to Git, a tool used by developers to manage and track changes in their code. This file tells Git which files or folders it should **ignore** and not include in the history of your project. In other words, it helps you keep your project clean and organized by excluding certain files or types of files.
```
[the remark this answers is not in the document]
Comment (SomeOrdinaryBro) [reply Before You Start/4. Git Basics.md:332]: Ty :smile:

[the remark this answers is not in the document]
Comment (SomeOrdinaryBro) [reply Before You Start/4. Git Basics.md:332]: ty :smile:👍

[the remark this answers is not in the document]
Comment (SomeOrdinaryBro) [reply Before You Start/4. Git Basics.md:332]: ty

**Pull request merged**